ci/cloudbuild/builds/cmake+gcc.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+#
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+export CC=gcc
+export CXX=g++
+
+cmake -GNinja -S . -B cmake-out
```

Comment:
For a future PR, the current CI scripts use a different binary output directory for each build (i.e. a different `-B` flag).  That allows us to run `./ci/kokoro/docker/build.sh asan` and `./ci/kokoro/docker/build.sh clang-tidy` in the same workspace, without mixing `.o` files from each build.
You may be able to keep these scripts simple by hiding the mount directory through some clever use of `--volume`.